framework/platform/bpl/uci/arp/monitor/arp_monitor.cpp

```diff
@@ -217,21 +219,21 @@ bool arp_monitor::start(std::string strIface)
     // open netlink sock
     if ((m_fdMonSocket = socket(AF_NETLINK, SOCK_RAW | SOCK_CLOEXEC, NETLINK_ROUTE)) < 0) {
         LOG(ERROR) << "Failed opening socket: " << strerror(errno);
-        return false;
+        return -int(eErrorCode::OPERATION_NOT_SUPPORTED);
     }
 
     int sndbuf = SOCKET_SEND_BUF_SIZE;
     if (setsockopt(m_fdMonSocket, SOL_SOCKET, SO_SNDBUF, &sndbuf, sizeof(sndbuf)) < 0) {
         LOG(ERROR) << "Failed setting SO_SNDBUF option: " << strerror(errno);
         stop();
-        return false;
+        return -1;
```

Comment:
Why not return the enum as you do in other places?

Reply:
`-1` represents general failure without any specific error code.  I used the enum to separate permissions error form other general errors. Basically bpl_err.h doesn't have a general error enum, and I wasn't sure if it is a good idea to add one. Anyway `1` is already taken for `UNKNOWN` on bpl_err.h, unless you would like to replace the `-1` with `-int(eErrorCode::UNKNOWN)`.